### ISSUE TITLE
Docker desktop freezes in starting state, and it won't run in version 4.8.2

### ISSUE THREAD

**Issue by Hiiirad** [docker/for-mac#6324]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: D2514F75-DADF-4004-AADB-7405979375FE/20220525074907

### Expected behavior

When I run the docker desktop program, I wanted it to be functional and responsive in short time.

### Actual behavior

The program freezes in "starting engine" state. I can't even change some settings. I can't use docker on my system!!!

### Information

  - macOS Version: 12.4
  - Intel chip or Apple chip: Apple Chip M1
  - Docker Desktop Version: 4.8.2

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[FAIL] DD0031: does the Docker API work? Cannot connect to the Docker daemon at unix://Library/Containers/com.docker.docker/Data/docker.raw.sock. Is the docker daemon running?
[FAIL] DD0004: is the Docker engine running? Get "http://ipc/docker": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:56:57.093932000Z][com.docker.diagnose][I] ipc.NewClient: 9601f8d5-com.docker.diagnose -> Library/Containers/com.docker.docker/Data/lifecycle-server.sock VMDockerdAPI
[linuxkit/pkg/desktop-host-tools/pkg/client.NewClientForPath(...)
[	linuxkit/pkg/desktop-host-tools/pkg/client/client.go:59
[linuxkit/pkg/desktop-host-tools/pkg/client.NewClient({0x1047da749, 0x13})
[	linuxkit/pkg/desktop-host-tools/pkg/client/client.go:53 +0xb0
[common/pkg/diagkit/gather/diagnose.isDockerEngineRunning()
[	common/pkg/diagkit/gather/diagnose/dockerd.go:21 +0x2c
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x104eb3fa0)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x44
[common/pkg/diagkit/gather/diagnose.Run.func1(0x104eb3fa0)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x44
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x104eb40a0?, 0x104eb3fa0)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x84
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x2, 0x104eb3fa0, 0x1400035f728)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x3c
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x104eb40a0?, 0x1400035f728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x20?, 0x1400035f728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
[common/pkg/diagkit/gather/diagnose.walkOnce(0x1049980a0?, 0x1400035f890)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0x8c
[common/pkg/diagkit/gather/diagnose.Run(0x104eb4320, 0x104265e40?, {0x1400035fb08, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x178
[main.checkCmd({0x140001b6010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:132 +0xe0
[main.main()
[	common/cmd/com.docker.diagnose/main.go:98 +0x308
[2022-05-25T07:56:57.098773000Z][com.docker.diagnose][I] (db5d9e19) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /docker
[2022-05-25T07:56:57.099292000Z][com.docker.diagnose][W] (db5d9e19) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /docker (515µs): Get "http://ipc/docker": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:56:57.099824000Z][com.docker.diagnose][I] (db5d9e19-1) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:56:57.100108000Z][com.docker.diagnose][W] (db5d9e19-1) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (282.667µs): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:56:58.100801000Z][com.docker.diagnose][I] (db5d9e19-2) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:56:58.103524000Z][com.docker.diagnose][W] (db5d9e19-2) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (2.704083ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:56:59.106935000Z][com.docker.diagnose][I] (db5d9e19-3) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:56:59.109571000Z][com.docker.diagnose][W] (db5d9e19-3) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (2.635458ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:57:00.112319000Z][com.docker.diagnose][I] (db5d9e19-4) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:57:00.115409000Z][com.docker.diagnose][W] (db5d9e19-4) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (3.077542ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:57:01.118258000Z][com.docker.diagnose][I] (db5d9e19-5) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:57:01.121478000Z][com.docker.diagnose][W] (db5d9e19-5) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (3.2035ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:57:02.127919000Z][com.docker.diagnose][I] (db5d9e19-6) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:57:02.129047000Z][com.docker.diagnose][W] (db5d9e19-6) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (1.124167ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:57:03.131504000Z][com.docker.diagnose][I] (db5d9e19-7) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:57:03.132828000Z][com.docker.diagnose][W] (db5d9e19-7) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (1.321416ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused
[2022-05-25T07:57:04.134632000Z][com.docker.diagnose][I] (db5d9e19-8) 9601f8d5-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2022-05-25T07:57:04.138253000Z][com.docker.diagnose][W] (db5d9e19-8) 9601f8d5-com.docker.diagnose C<-S NoResponse GET /ping (3.610708ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/lifecycle-server.sock: connect: connection refused

[FAIL] DD0011: are the LinuxKit services running? failed to ping VM diagnosticsd with error: Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:04.141358000Z][com.docker.diagnose][I] ipc.NewClient: 19460154-diagnose -> Library/Containers/com.docker.docker/Data/diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.glob..func12()
[	common/pkg/diagkit/gather/diagnose/linuxkit.go:18 +0x9c
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x104eb3f20)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x44
[common/pkg/diagkit/gather/diagnose.Run.func1(0x104eb3f20)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x44
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x104eb3fa0?, 0x104eb3f20)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x84
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x3, 0x104eb3f20, 0x1400061f728)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x3c
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x2, 0x104eb3fa0?, 0x1400061f728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x104eb40a0?, 0x1400061f728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x20?, 0x1400061f728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
[common/pkg/diagkit/gather/diagnose.walkOnce(0x1049980a0?, 0x1400035f890)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0x8c
[common/pkg/diagkit/gather/diagnose.Run(0x104eb4320, 0x104265e40?, {0x1400035fb08, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x178
[main.checkCmd({0x140001b6010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:132 +0xe0
[main.main()
[	common/cmd/com.docker.diagnose/main.go:98 +0x308
[2022-05-25T07:57:04.145554000Z][com.docker.diagnose][I] (dff51620) 19460154-diagnose C->S diagnosticsd GET /ping
[2022-05-25T07:57:04.145898000Z][com.docker.diagnose][W] (dff51620) 19460154-diagnose C<-S NoResponse GET /ping (339.958µs): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused

[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
Error response from daemon: dial unix /Users/hirad/Library/Containers/com.docker.docker/Data/docker.raw.sock: connect: connection refused
[FAIL] DD0003: is the Docker CLI working? exit status 1
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[FAIL] DD0014: are the backend processes running? 2 errors occurred:
	* querying com.docker.driver.amd64-linux process: is it running as a different user?: no such file or directory
	* querying qemu-system-aarch64 process: is it running as a different user?: no such file or directory


[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[FAIL] DD0012: is the VM networking working? network checks failed: Post "http://ipc/check-network-connectivity": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:05.957187000Z][com.docker.diagnose][I] ipc.NewClient: f9500f74-diagnose-network -> Library/Containers/com.docker.docker/Data/diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[	common/pkg/diagkit/gather/diagnose/network.go:34 +0xd4
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x104eb41a0)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x44
[common/pkg/diagkit/gather/diagnose.Run.func1(0x104eb41a0)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x44
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2?, 0x104eb41a0)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x84
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x104eb41a0, 0x14000587728)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x3c
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x20?, 0x14000587728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x78
[common/pkg/diagkit/gather/diagnose.walkOnce(0x1049980a0?, 0x1400035f890)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0x8c
[common/pkg/diagkit/gather/diagnose.Run(0x104eb4320, 0x104265e40?, {0x1400035fb08, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x178
[main.checkCmd({0x140001b6010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:132 +0xe0
[main.main()
[	common/cmd/com.docker.diagnose/main.go:98 +0x308
[2022-05-25T07:57:05.959607000Z][com.docker.diagnose][I] (8d10bf14) f9500f74-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["192.168.1.100","10.5.0.2"]}
[2022-05-25T07:57:05.960063000Z][com.docker.diagnose][W] (8d10bf14) f9500f74-diagnose-network C<-S NoResponse POST /check-network-connectivity (498µs): Post "http://ipc/check-network-connectivity": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:05.960641000Z][com.docker.diagnose][I] (8d10bf14-1) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:05.960919000Z][com.docker.diagnose][W] (8d10bf14-1) f9500f74-diagnose-network C<-S NoResponse GET /ping (276.375µs): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:06.961839000Z][com.docker.diagnose][I] (8d10bf14-2) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:06.964760000Z][com.docker.diagnose][W] (8d10bf14-2) f9500f74-diagnose-network C<-S NoResponse GET /ping (2.905333ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:07.968920000Z][com.docker.diagnose][I] (8d10bf14-3) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:07.971527000Z][com.docker.diagnose][W] (8d10bf14-3) f9500f74-diagnose-network C<-S NoResponse GET /ping (2.602875ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:08.974215000Z][com.docker.diagnose][I] (8d10bf14-4) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:08.976843000Z][com.docker.diagnose][W] (8d10bf14-4) f9500f74-diagnose-network C<-S NoResponse GET /ping (2.617375ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:09.980005000Z][com.docker.diagnose][I] (8d10bf14-5) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:09.982562000Z][com.docker.diagnose][W] (8d10bf14-5) f9500f74-diagnose-network C<-S NoResponse GET /ping (2.549417ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:10.985292000Z][com.docker.diagnose][I] (8d10bf14-6) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:10.988451000Z][com.docker.diagnose][W] (8d10bf14-6) f9500f74-diagnose-network C<-S NoResponse GET /ping (3.147625ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:11.991338000Z][com.docker.diagnose][I] (8d10bf14-7) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:11.993953000Z][com.docker.diagnose][W] (8d10bf14-7) f9500f74-diagnose-network C<-S NoResponse GET /ping (2.607167ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused
[2022-05-25T07:57:12.998031000Z][com.docker.diagnose][I] (8d10bf14-8) f9500f74-diagnose-network C->S diagnosticsd GET /ping
[2022-05-25T07:57:12.999649000Z][com.docker.diagnose][W] (8d10bf14-8) f9500f74-diagnose-network C<-S NoResponse GET /ping (1.610417ms): Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused

[FAIL] DD0032: do Docker networks overlap with host IPs? Cannot connect to the Docker daemon at unix://Library/Containers/com.docker.docker/Data/docker.raw.sock. Is the docker daemon running?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 3 issues:

1 : The test: are the LinuxKit services running?
    Failed with: failed to ping VM diagnosticsd with error: Get "http://ipc/ping": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused

The Docker engine runs inside a Linux VM as a service. Therefore the services must have started.

2 : The test: are the backend processes running?
    Failed with: 2 errors occurred:
	* querying com.docker.driver.amd64-linux process: is it running as a different user?: no such file or directory
	* querying qemu-system-aarch64 process: is it running as a different user?: no such file or directory



Not all of the backend processes are running.

3 : The test: is the VM networking working?
    Failed with: network checks failed: Post "http://ipc/check-network-connectivity": dial unix Library/Containers/com.docker.docker/Data/diagnosticd.sock: connect: connection refused

VM seems to have a network connectivity issue. Please check your host firewall and anti-virus settings in case they are blocking the VM.

```

### Steps to reproduce the behavior

  1. Install Docker Desktop
  2. Run the program 
  3. It freezes in starting process and it takes so much RAM, and even swap from the system.

**Comment by chuck-g** [comment]
I'm having the same issue on my M1 Max. Same version of OSX and Docker. 

To clarify, the `qemu-system-aarch64` process is consuming 16GB of memory (12GB real memory).

**Comment by gkbrown22** [comment]
Similar issue on my Intel-based mac Forever in startup mode. Reinstalled, restarted etc. No joy.

**Comment by aXent** [comment]
I'm experiencing the same issue (Also M1 apple chip - M1 Max). Same OSX and Docker version. Diagnostics output is almost identical.

**Comment by stunami** [comment]
Same issue for me as well, M1 Pro, OSX 12.4 (Just upgraded) and docker desktop for mac 4.8.2 (79419)

**Comment by cbake6807** [comment]
Same here, Intel Chip MacBook Pro (16-inch, 2019). Just upgraded from 11.x to 12.5. Everything seemed ok, but i ran the DockerDesktop Software Update, and then it died. Most links worked except the Containers page. That one hung and froze the app. I ran the purge data and now i can access the Containers page. Have to rebuild my containers again. :)

**Comment by Hiiirad** [comment]
This issue was fixed on Docker Desktop version 4.9.0
